qiskit/quantum_info/operators/symplectic/pauli_utils.py

```diff
@@ -13,40 +13,24 @@
 PauliList utility functions.
 """
 
-import warnings
-
-import numpy as np
-
 from qiskit.quantum_info.operators.symplectic.pauli_list import PauliList
-from qiskit.quantum_info.operators.symplectic.pauli_table import PauliTable
+from qiskit.utils.deprecation import deprecate_arguments
 
 
-def pauli_basis(num_qubits, weight=False, pauli_list=False):
+# pylint: disable=missing-param-doc,missing-type-doc,unused-argument
+@deprecate_arguments({"pauli_list": "_"})
```

Comment:
Q: is this correct way to deprecate arguments? just hacky...

Reply:
Maybe we could extend `deprecate_arguments` to accept a list of arguments that are deprecated without replacement, but right now your solution seems sensible to me 🤔